### PR TITLE
Update to recipe to post2 release and fix permissions of .egg-info directory

### PR DESCRIPTION
**Pull request by jjhelmus**

### CODE REVIEW

**Comment by jjhelmus** [comment]
commit b302c79 updates this recipe to the [post2 release from PyPI](https://pypi.python.org/pypi/protobuf/3.0.0b2.post2).  I'm not sure what different exist between this release and the earlier releases but pip will install the latest post release.

commit b776772 address the permissions of the .egg-info directory which can cause issues on multi-user systems where a single conda directory is shared between multiple users.

The best explanation I can give for this is showing the permissions difference between the two directory in the produced bz2.

```
[root@evs348532 foo]# git rev-parse HEAD
b302c7913286c340dad595d419509f7d3a56c805
[root@evs348532 foo]# conda build recipe
...
[root@evs348532 foo]# tar tvf /opt/conda/conda-bld/linux-64/protobuf-3.0.0b2-py27_1.tar.bz2 
-rw-r--r-- root/root      6179 2016-03-11 16:15 info/files
...
-rw-r--r-- root/root       305 2016-03-11 16:15 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7-nspkg.pth
-rw-r----- root/root       668 2016-03-11 16:15 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7.egg-info/PKG-INFO
-rw-r----- root/root      1807 2016-03-11 16:15 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7.egg-info/SOURCES.txt
-rw-r----- root/root         1 2016-03-11 16:15 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7.egg-info/dependency_links.txt
-rw-r----- root/root         7 2016-03-11 16:15 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7.egg-info/namespace_packages.txt
-rw-r----- root/root        20 2016-03-11 16:15 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7.egg-info/requires.txt
-rw-r----- root/root         7 2016-03-11 16:15 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7.egg-info/top_level.txt

[root@evs348532 foo]# git rev-parse HEAD
b776772aa6475ceaac6d1091c53286e03749da76
[root@evs348532 foo]# conda build recipe
...
[root@evs348532 foo]# tar tvf /opt/conda/conda-bld/linux-64/protobuf-3.0.0b2-py27_1.tar.bz2 
-rw-r--r-- root/root      6179 2016-03-11 16:18 info/files
...
-rw-r--r-- root/root       305 2016-03-11 16:18 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7-nspkg.pth
-rw-r--r-- root/root       668 2016-03-11 16:18 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7.egg-info/PKG-INFO
-rw-r--r-- root/root      1807 2016-03-11 16:18 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7.egg-info/SOURCES.txt
-rw-r--r-- root/root         1 2016-03-11 16:18 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7.egg-info/dependency_links.txt
-rw-r--r-- root/root         7 2016-03-11 16:18 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7.egg-info/namespace_packages.txt
-rw-r--r-- root/root        20 2016-03-11 16:18 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7.egg-info/requires.txt
-rw-r--r-- root/root         7 2016-03-11 16:18 lib/python2.7/site-packages/protobuf-3.0.0b2.post2-py2.7.egg-info/top_level.txt
```

I don't see an issue with egg-info directory permissions on the current bz2 files on conda-forge (-rw-r--r-- root/root). I can't really explain this since when I build the package in the same Docker container I get different permissions (-rw-r----- root/root)

**Comment by dopplershift** [comment]
Regarding post2, I think it's google/protobuf#1307.

As far as permissions go, could the container be inheriting a umask or something from outside? Or maybe different versions of the docker engine?

**Comment by jjhelmus** [comment]
I've been trying to nail down where the permissions of the .egg-info directory gets set but I got overwhelmed trying to control all the possible candidates  (umask of the docker host and container, permissions of files in the various tarballs, the version of conda, conda-build, setuptools, Docker).  

The chmod command seemed to me to be a kludgy but explicit way to solve the issue but if someone wants to look into this further, I'd be happy to have another set of eyes on the problem.  It is also quite possible that the chmod is completely unnecessary for the conda-forge setup. I can revert the change and check the permissions on the conda package produced.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -5,14 +5,12 @@ package:
     version: {{ version }}
 
 source:
-    fn: protobuf-{{ version }}.tar.gz
-    url: https://pypi.python.org/packages/source/p/protobuf/protobuf-{{ version }}.tar.gz
-    md5: f0d3bd2394345a9af4a277cd0302ae83
+    fn: protobuf-{{ version }}.post2.tar.gz
+    url: https://pypi.python.org/packages/source/p/protobuf/protobuf-{{ version }}.post2.tar.gz
```
Comment (dopplershift) [inline recipe/meta.yaml]: I think the `.post2` should go in the declaration of the `version` variable at the top of the file.

**Comment by dopplershift** [comment]
I think the .egg-info permissions come from the original tarball:
```
>tar tvf ~/Downloads/protobuf-3.0.0b2.post2.tar.gz
drwxr-x---  0 silviuc eng         0 Mar  7 09:05 protobuf-3.0.0b2.post2/
drwxr-x---  0 silviuc eng         0 Mar  7 09:05 protobuf-3.0.0b2.post2/protobuf.egg-info/
-rw-r-----  0 silviuc eng      1797 Mar  7 09:05 protobuf-3.0.0b2.post2/protobuf.egg-info/SOURCES.txt
-rw-r-----  0 silviuc eng         7 Mar  7 09:05 protobuf-3.0.0b2.post2/protobuf.egg-info/top_level.txt
-rw-r-----  0 silviuc eng         1 Mar  7 09:05 protobuf-3.0.0b2.post2/protobuf.egg-info/dependency_links.txt
-rw-r-----  0 silviuc eng        20 Mar  7 09:05 protobuf-3.0.0b2.post2/protobuf.egg-info/requires.txt
-rw-r-----  0 silviuc eng       668 Mar  7 09:05 protobuf-3.0.0b2.post2/protobuf.egg-info/PKG-INFO
-rw-r-----  0 silviuc eng         7 Mar  7 09:05 protobuf-3.0.0b2.post2/protobuf.egg-info/namespace_packages.txt
-rw-r-----  0 silviuc eng        59 Mar  7 09:05 protobuf-3.0.0b2.post2/setup.cfg
-rw-r-----  0 silviuc eng       668 Mar  7 09:05 protobuf-3.0.0b2.post2/PKG-INFO
-rwxr-x---  0 silviuc eng      9114 Mar  7 08:38 protobuf-3.0.0b2.post2/setup.py
drwxr-x---  0 silviuc eng         0 Mar  7 09:05 protobuf-3.0.0b2.post2/google/
```
I'm not sure if this argues for anything other than your script, though.

**Comment by jjhelmus** [comment]
@dopplershift 

> I think the .egg-info permissions come from the original tarball:

That is my hunch also, but the non-post2 tarball has the same permissions (but a different user) but the conda package on Anaconda.org has this directory world readable which I can't explain.  Regardless I don't see the chmod hurting anything even if it is unnecessary.

**Comment by dopplershift** [comment]
I agree about the chmod--looks good to me!

**Pull request merged**